fraos/src/seqno.rs

```diff
@@ -38,19 +49,64 @@ impl SeqNoIndex {
     }
 
     /// Get the location of a record with the given number.
-    pub fn get_pointer_to_value(&self, seqno: usize) -> Option<u64> {
-        let offset = seqno * size_of::<u64>();
+    pub fn get_offset_and_length(
+        &self,
+        seqno: usize,
+    ) -> Result<Option<(usize, usize)>, FraosError> {
+        let offset = seqno * Self::SIZE_OF_USIZE * 2;
+
+        match self.inner.get_data(offset, |mmap| {
+            let mut offset_buffer = [0u8; Self::SIZE_OF_USIZE];
+            let mut length_buffer = [0u8; Self::SIZE_OF_USIZE];
+            offset_buffer.copy_from_slice(&mmap[..Self::SIZE_OF_USIZE]);
+            length_buffer.copy_from_slice(&mmap[Self::SIZE_OF_USIZE..Self::SIZE_OF_USIZE * 2]);
+
+            Ok(Some((
+                usize::from_le_bytes(offset_buffer),
+                usize::from_le_bytes(length_buffer),
+            )))
+        })? {
+            None => Ok(None),
+            Some(results) => Ok(Some(results)),
+        }
+    }
 
-        self.inner.get_data(offset, |mmap| {
-            let mut key_length_bytes = [0u8; size_of::<u64>()];
-            key_length_bytes.copy_from_slice(&mmap[..size_of::<u64>()]);
+    pub fn len(&self) -> usize {
+        self.inner.memory_size() / Self::SIZE_OF_USIZE / 2
+    }
 
-            Some(u64::from_le_bytes(key_length_bytes))
-        })
+    #[allow(unused)]
+    pub fn memory_size(&self) -> usize {
+        self.inner.memory_size()
     }
 
-    pub fn size(&self) -> usize {
-        self.inner.memory_size() / size_of::<u64>()
+    pub fn is_correct(&self) -> Result<(), FraosError> {
+        if self.inner.memory_size() != self.len() * Self::SIZE_OF_USIZE * 2 {
```

Comment:
does this actually check anything? why would it fail? it's inverting what `self.len()` does by definition.

Reply:
isn't it more clear to check that `self.inner.memory_size() % (Self::SIZE_OF_USIZE * 2) != 0` then? Is that what's being done here?